src/vmm/src/guest_config/templates_serde.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+
+use serde::de::Error as SerdeError;
+use serde::{Deserialize, Deserializer, Serializer};
+
+use crate::guest_config::templates::RegisterValueFilter;
+
+macro_rules! deserialize_reg_value_filter {
```

Comment:
I'm sure there is a reason for that, but why can't this be achieved with a generic function?

Reply:
The use of `from_str_radix` makes is hard, because it can it is not a part of any trait.
We can implement some `NumericTrait` for all numeric types that have all needed details in it. With it we can drop all these separate functions and just implement `Serialize` and `Deserialize` on `RegisterValueFilter` directly. (Will need to redo serialization function though, to iterate over bits directly instead of relying on formatting first)
The only problem I see with this approach is where should this `NumericTrait` exist? It will be very generic and maybe useful in other parts of the codebase.

Reply:
Added another commit with another implementation for ser/de for reference. I kinda like it. Let me know your thoughts about it.